cmd/car/car.go

```diff
@@ -231,6 +231,29 @@ func main1() int {
 					},
 				},
 			},
+			{
+				Name:    "put-block",
+				Aliases: []string{"pb"},
+				Usage:   "Put a block into a car",
+				Action:  PutCarBlock,
+				Flags: []cli.Flag{
+					&cli.StringFlag{
+						Name:    "codec",
+						Aliases: []string{"c"},
+						Usage:   "Codec to use for CID generation (raw, dag-pb, dag-cbor, dag-json)",
```

Comment:
in the use case examples you provided, i wonder if there is an 'auto-detect' that's worthwhile in trying to auto 'upgrade' to the standard json/cbor/pb when data can be parsed as such - or if there's a way to get this from the cid rather than potentially loosing it?

Reply:
for the _very specific_ usecase yes, we could autodetect it, but I feel like requiring the cid as well as contents of the block will be annoying in the more general case; we could accept a `--cid` parameter, with either codec or cid needing to be set